File: backend/app/Models/Doctor.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Doctor extends Model
{
    use HasFactory;

    // Define the table name explicitly (optional if it matches the model name in plural form)
    protected $table = 'doctors';

    // The primary key for the table
    protected $primaryKey = 'doctor_id';

    // Disable auto-incrementing for the primary key (if necessary)
    public $incrementing = true;

    // The attributes that are mass assignable
    protected $fillable = [
        'first_name',
        'last_name',
        'specialization',
        'email',
        'phone_number',
        'department',
        'birthdate',
        'address',
        'status',
        'notes'
    ];

    // Set the date format for the model
    const CREATED_AT = 'created_at';
    const UPDATED_AT = 'updated_at';

    // Use custom date format if needed
    protected $casts = [
        'birthdate' => 'date',
    ];
}

File: backend/app/models/doctor.py
#app/models/doctor.py

from sqlalchemy import Column, Integer, String, Boolean, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.database import Base
from datetime import datetime

class Doctor(Base):
    __tablename__ = 'doctors'

    id = Column(Integer, primary_key=True, index=True)
    # Link to users table (authentication)
    user_id = Column(Integer, ForeignKey('users.id'), nullable=False)
    first_name = Column(String, nullable=False)
    last_name = Column(String, nullable=False)

    specialty = Column(String, nullable=False)
    license_number = Column(String, unique=True, nullable=False)

    phone_number = Column(String, nullable=False)
    email = Column(String, unique=True, index=True, nullable=False)

    experince_years = Column(Integer, nullable=False)
    is_active = Column(Boolean, default=True)

    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    # Relationship with Patient model   
    patients = relationship("Patient", back_populates="doctor")
    user = relationship("User", back_populates="doctor_profile")
    appointments = relationship("Appointment", back_populates="doctor", )
    prescriptions = relationship("Prescription", back_populates="doctor")

    def __repr__(self):
        return f"<Doctor(id={self.id}, name={self.first_name} {self.last_name}, specialty={self.specialty})>"
    
